src/dotnet/UI.Blazor/Components/ContextMenu/BlazorContextMenu/Components/ContextMenuBase.cs

```diff
@@ -0,0 +1,263 @@
+using BlazorContextMenu.Services;
+
+namespace BlazorContextMenu;
+
+public abstract class ContextMenuBase : MenuTreeComponent
+{
+    [Inject] private BlazorContextMenuSettings Settings { get; set; } = null!;
+    [Inject] private IContextMenuStorage ContextMenuStorage { get; set; } = null!;
+    [Inject] private IInternalContextMenuHandler ContextMenuHandler { get; set; } = null!;
+    [Inject] private IJSRuntime JSRuntime { get; set; } = null!;
+    [Inject] private IMenuTreeTraverser MenuTreeTraverser { get; set; } = null!;
+
+    [Parameter(CaptureUnmatchedValues = true)]
+    public Dictionary<string, object> Attributes { get; set; } = null!;
+
+    protected virtual string BaseClass => "blazor-context-menu blazor-context-menu__wrapper";
+
+    /// <summary>
+    /// The id that the <see cref="ContextMenuTrigger" /> will use to bind to. This parameter is required
+    /// </summary>
+    [Parameter]
+    public string Id { get; set; } = "";
+
+    /// <summary>
+    /// The name of the template to use for this <see cref="ContextMenu" /> and all its <see cref="SubMenu" />.
+    /// </summary>
+    [Parameter]
+    public string? Template { get; set; }
+
+    [CascadingParameter(Name = "CascadingTemplate")]
+    protected string? CascadingTemplate { get; set; }
+
+    /// <summary>
+    /// Allows you to override the default css class of the <see cref="ContextMenu"/>'s div element, for full customization.
+    /// </summary>
+    [Parameter]
+    public string? OverrideDefaultCssClass { get; set; }
+
+    /// <summary>
+    /// Allows you to override the default css class of the <see cref="ContextMenu"/>'s div element while it's shown, for full customization.
+    /// </summary>
+    [Parameter]
+    public string? OverrideDefaultShownCssClass { get; set; }
+
+    /// <summary>
+    /// Allows you to override the default css class of the <see cref="ContextMenu"/>'s div element while it's hidden, for full customization.
+    /// </summary>
+    [Parameter]
+    public string? OverrideDefaultHiddenCssClass { get; set; }
+
+    /// <summary>
+    /// Allows you to override the default css class of the <see cref="ContextMenu"/>'s ul element, for full customization.
+    /// </summary>
+    [Parameter]
+    public string? OverrideDefaultListCssClass { get; set; }
+
+    /// <summary>
+    /// Additional css class that is applied to the <see cref="ContextMenu"/>'s div element. Use this to extend the default css.
+    /// </summary>
+    [Parameter]
+    public string? CssClass { get; set; }
```

Comment:
Ok, I get the problem now - you want to either add these classes or override. One option here is to add "--" helper class, which is replaced by common routine like `CssClasses.Combine(string @base, string extra)`, which simply discards `@base` if `extra` starts with `"-- "`.

Reply:
And we could use this approach everywhere in future.

Reply:
Simplified